#65 - toggle password visibility

Lee0z opened the pull request:
Added button to toggle password visibility.

- Invisible:
![image](https://github.com/blumilksoftware/escooters/assets/102246039/159824bf-1d5d-4d69-a535-db33782c0a5e)


- Visible:
![image](https://github.com/blumilksoftware/escooters/assets/102246039/67604da5-ba87-49b0-bd5a-c2d6116861c0)

Same goes to register form.

- Invisible:
![image](https://github.com/blumilksoftware/escooters/assets/102246039/0f0fda6e-76bf-486e-b8e8-47a8d30299c1)

- Visible:
![image](https://github.com/blumilksoftware/escooters/assets/102246039/af118c87-901b-45d2-b9ef-fc1902897f08)

It should close #65.

vojcc commented on resources/js/Shared/Layout/Nav.vue:
```diff
@@ -204,15 +213,18 @@ defineExpose({
+              <input v-model="registerForm.password" :type="isPasswordVisible ? 'text' : 'password'"
                      class="w-full rounded-lg border-blumilk-200 py-3 md:p-2" required
               >
+              <button type="button" class="absolute bottom-2 right-2" @click="togglePasswordVisibility">
```
```suggestion
              <button type="button" class="absolute bottom-3 right-2 md:bottom-2" @click="togglePasswordVisibility">
```

vojcc replied:
That would look better on mobile devices.

EwelinaSkrzypacz reviewed (approved):
I think I will get rid of filed "password confirmation" because now we have toggle to see password, but maybe it's a task for another pr?

Lee0z commented:
> I think I will get rid of filed "password confirmation" because now we have toggle to see password, but maybe it's a task for another pr?

So the button in registration form doesn't reveal "Confirm Password" field? Just the "Password" one?

EwelinaSkrzypacz commented:
> > I think I will get rid of filed "password confirmation" because now we have toggle to see password, but maybe it's a task for another pr?
> 
> So the button in registration form doesn't reveal "Confirm Password" field? Just the "Password" one?


No, I mean that we should delete input "Confirm password". This field is unnecessary due to potential user errors, the inconvenience of entering complex passwords twice, the availability of more advanced verification methods, and the desire for a smoother user experience. 


![image](https://github.com/blumilksoftware/escooters/assets/56546832/aa4e2045-b774-49b6-9db7-9331faabba49)

Lee0z commented:
> > > I think I will get rid of filed "password confirmation" because now we have toggle to see password, but maybe it's a task for another pr?
> > 
> > 
> > So the button in registration form doesn't reveal "Confirm Password" field? Just the "Password" one?
> 
> No, I mean that we should delete input "Confirm password". This field is unnecessary due to potential user errors, the inconvenience of entering complex passwords twice, the availability of more advanced verification methods, and the desire for a smoother user experience.
> 
> ![image](https://user-images.githubusercontent.com/56546832/281377048-aa4e2045-b774-49b6-9db7-9331faabba49.png)

That is great idea, so I changed registration form:
<img width="236" alt="image" src="https://github.com/blumilksoftware/escooters/assets/102246039/6e8f38b5-3801-40ad-bcbb-064b8f6e1397">

Is there any backend code to support that password confirmation? As far as I know it is only frontend feature. Please check commit after this comment.

EwelinaSkrzypacz commented:
> > > > I think I will get rid of filed "password confirmation" because now we have toggle to see password, but maybe it's a task for another pr?
> > > 
> > > 
> > > So the button in registration form doesn't reveal "Confirm Password" field? Just the "Password" one?
> > 
> > 
> > No, I mean that we should delete input "Confirm password". This field is unnecessary due to potential user errors, the inconvenience of entering complex passwords twice, the availability of more advanced verification methods, and the desire for a smoother user experience.
> > ![image](https://user-images.githubusercontent.com/56546832/281377048-aa4e2045-b774-49b6-9db7-9331faabba49.png)
> 
> That is great idea, so I changed registration form: <img alt="image" width="236" src="https://user-images.githubusercontent.com/102246039/281385865-6e8f38b5-3801-40ad-bcbb-064b8f6e1397.png">
> 
> Is there any backend code to support that password confirmation? As far as I know it is only frontend feature. Please check commit after this comment.

In files there are some use of `password_confirmation`, for example

- `Feature/SignupTest.php`
- `Exceptions/ExceptionHandler.php`
- `Http/Middleware/TrimStrings.php`

I will recommend use `Ctrl + Shift + F` in Phpstorm to search all of occurrences of text

Lee0z commented:
> > > > > I think I will get rid of filed "password confirmation" because now we have toggle to see password, but maybe it's a task for another pr?
> > > > 
> > > > 
> > > > So the button in registration form doesn't reveal "Confirm Password" field? Just the "Password" one?
> > > 
> > > 
> > > No, I mean that we should delete input "Confirm password". This field is unnecessary due to potential user errors, the inconvenience of entering complex passwords twice, the availability of more advanced verification methods, and the desire for a smoother user experience.
> > > ![image](https://user-images.githubusercontent.com/56546832/281377048-aa4e2045-b774-49b6-9db7-9331faabba49.png)
> > 
> > 
> > That is great idea, so I changed registration form: <img alt="image" width="236" src="https://user-images.githubusercontent.com/102246039/281385865-6e8f38b5-3801-40ad-bcbb-064b8f6e1397.png">
> > Is there any backend code to support that password confirmation? As far as I know it is only frontend feature. Please check commit after this comment.
> 
> In files there are some use of `password_confirmation`, for example
> 
> * `Feature/SignupTest.php`
> * `Exceptions/ExceptionHandler.php`
> * `Http/Middleware/TrimStrings.php`
> 
> I will recommend use `Ctrl + Shift + F` in Phpstorm to search all of occurrences of text

There was indeed written tests for signup feature, so I've removed password_confirmation field.
Since we don't need "password_confirmation" to be trimmed through middleware, I've deleted that here too, same goes to exceptions.

In backend register request there were also rules specified, which required "confirmation", deleted.

I've manualy tested account creation few times and after modifications it worked. 

Also noticed small bug - on error event (wrong password input etc.) the password visibility button moved down to match both error and password field. It's fixed.
